### PR TITLE
Persistence username storage + tabs

### PR DESCRIPTION
**Pull request by r00t-admin**
Create UI screen for Minerva Extension. Persistent username storage.

Items:
1. User can enter their name into text field and name will be stored in chrome.storage.sync.
2. Added tabs for students to view their personal time watched, rankings, and store.

### CODE REVIEW

**Comment by Yuliy97** [comment]
can  u add a screenshot of what the ui looks  like now?

**Comment by r00t-admin** [comment]
image

**Comment by r00t-admin** [comment]
image

**Comment by r00t-admin** [comment]
image

**Comment by Yuliy97** [comment]
can u make the items same size, why  is the star so much bigger

**Review by Jason-KChen — CHANGES_REQUESTED** [review]
see comments

File: extension/manifest.json [hunk 1]
```diff
@@ -13,6 +18,11 @@
       "js": ["src/content_scripts/comm.js", "src/content_scripts/StudyModeLocker.js"]
     }
   ],
-  "web_accessible_resources": ["style/style.css"],
+  "background": {
+    "scripts": ["src/background_scripts/background.js"],
```
Comment (Jason-KChen) [inline extension/manifest.json:26]: good

File: extension/src/background_scripts/background.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// eslint-disable-next-line no-undefined
+chrome.runtime.onInstalled.addListener(function() {
```
Comment (Jason-KChen) [inline extension/src/background_scripts/background.js:1]: this should no longer report error on next push

File: extension/src/background_scripts/background.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// eslint-disable-next-line no-undefined
```
Comment (Jason-KChen) [inline extension/src/background_scripts/background.js]: this is close, should be `no-undef`
anyways, no longer need (see my review comments)

File: extension/src/content_scripts/GoogleClassroom.js [hunk 4]
```diff
@@ -57,4 +57,3 @@ window.addEventListener("load", (_) => {
     window.addEventListener("click", hijackYoutubeLinkClicks);
     console.log("Google Classroom Handler registered");
 });
-
```
Comment (Jason-KChen) [inline extension/src/content_scripts/GoogleClassroom.js]: Should be handled by `EditorConfig` extension automatically if you install it

File: extension/src/popups/MinervaMenu.html [hunk 5]
```diff
@@ -0,0 +1,101 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <style>
+      input {
+        height: 30px;
```
Comment (Jason-KChen) [inline extension/src/popups/MinervaMenu.html]: Put all inline `style` tag into another css file please

File: extension/src/popups/MinervaMenu.html [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+              <td>Otto</td>
+              <td>14:12</td>
+            </tr>
+            <tr>
+              <th scope="row">3</th>
+              <td>Jacob</td>
+              <td>Thornton</td>
+              <td>1:2</td>
+            </tr>
+
```
Comment (Jason-KChen) [inline extension/src/popups/MinervaMenu.html]: nit: remove

File: extension/src/popups/MinervaMenu.html [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+          </tbody>
+        </table>
+      </div>
+      <div class="tab-pane fade" id="nav-store" role="tabpanel" aria-labelledby="nav-store-tab">
+        Hello students! This is what have for sale!
+        <table id="store-table"class="table">
+          <thead>
+            <tr>
+
```
Comment (Jason-KChen) [inline extension/src/popups/MinervaMenu.html:95]: nit: remove

File: extension/src/popups/MinervaMenu.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable no-undefined */
```
Comment (Jason-KChen) [inline extension/src/popups/MinervaMenu.js]: see my comments above

File: extension/src/popups/MinervaMenu.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+window.onload = function() {
+    let studentNameField = document.getElementById("studentName");
+    studentNameField.onkeydown = function(event) {
+        if (event.code == "Enter" || event.key == "Enter") {
+            chrome.tabs.query({currentWindow: true, active: true}, function(tabs) {
+                console.log(studentNameField.value);
+                chrome.storage.local.set({"studentName": studentNameField.value}, function() {
+                    chrome.storage.local.get(console.log);
```
Comment (Jason-KChen) [inline extension/src/popups/MinervaMenu.js]: what is this line for?

File: extension/src/popups/MinervaMenu.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable no-undefined */
+window.onload = function() {
+    let studentNameField = document.getElementById("studentName");
+    studentNameField.onkeydown = function(event) {
+        if (event.code == "Enter" || event.key == "Enter") {
+            chrome.tabs.query({currentWindow: true, active: true}, function(tabs) {
```
Comment (Jason-KChen) [inline extension/src/popups/MinervaMenu.js]: why do we need the `tabs` API? is there a reason we end up not using `tabs` variable?

File: extension/src/popups/MinervaMenu.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+                console.log(studentNameField.value);
+                chrome.storage.local.set({"studentName": studentNameField.value}, function() {
+                    chrome.storage.local.get(console.log);
+                });
+
```
Comment (Jason-KChen) [inline extension/src/popups/MinervaMenu.js]: nit: remove

File: extension/src/popups/MinervaMenu.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+    let images = ["golden_star.jpg", "pencil.jpg", "ruler.jpg"];
+    for (var i = 1; i <= 3; i++) {
+        let image = document.getElementById(`item-${i}-image`);
+        let imgURL = chrome.extension.getURL(`images/${images[i - 1]}`);
+        image.src = imgURL;
+        image.width = "50";
+        image.height = "50";
+    }
+};
+/* eslint-enable no-undefined */
```
Comment (Jason-KChen) [inline extension/src/popups/MinervaMenu.js]: nit: remove

File: src/MinervaMenu.html [hunk 13]
```diff
@@ -0,0 +1,106 @@
+<!DOCTYPE html>
```
Comment (Jason-KChen) [inline src/MinervaMenu.html]: This file should not exist?

File: src/MinervaMenu.js [hunk 14]
```diff
@@ -0,0 +1,26 @@
+window.onload = function() {
```
Comment (Jason-KChen) [inline src/MinervaMenu.js]: This file should not exist?

File: src/background.js [hunk 15]
```diff
@@ -0,0 +1,10 @@
+chrome.runtime.onInstalled.addListener(function() {
```
Comment (Jason-KChen) [inline src/background.js]: This file should not exist?

**Review by Jason-KChen — CHANGES_REQUESTED** [review]
Looks like merge conflict resolution did not go too well but the code style is much better this time.

I added `chrome` as a global variable from this point, you don't have to worry about it being a `undeclared` variable from now.

Please re-request review when ready by clicking on the circle next to my name next time. 

Also, please address my comments as well (resolve/reply)

Let's aim to merge this soon...

File: extension/manifest.json (hunk 1 above)
Comment (r00t-admin) [reply extension/manifest.json:26]: ok

File: extension/src/background_scripts/background.js (hunk 2 above)
Comment (r00t-admin) [reply extension/src/background_scripts/background.js:1]: resolved

**Review by Jason-KChen — APPROVED** [review]

**Pull request merged**